pkgs/applications/editors/jetbrains/source/build.nix

```
@@ -209,27 +222,36 @@ stdenvNoCC.mkDerivation rec {
   patches = [
     ../patches/no-download.patch
     ../patches/disable-sbom-generation.patch
+    ../patches/bump-jackson-core-in-source.patch
   ];
 
   postPatch = ''
     cp ${restarter}/bin/restarter bin/linux/amd64/restarter
     cp ${fsnotifier}/bin/fsnotifier bin/linux/amd64/fsnotifier
     cp ${libdbm}/lib/libdbm.so bin/linux/amd64/libdbm.so
 
-    sed \
-      -e 's|JPS_PLUGIN_CLASSPATH_HERE|${kotlin-jps-plugin-classpath}|' \
-      -e 's|KOTLIN_PATH_HERE|${kotlin}|' \
-      -i platform/build-scripts/src/org/jetbrains/intellij/build/kotlin/KotlinCompilerDependencyDownloader.kt
-    sed \
-      -e 's|JDK_PATH_HERE|${jbr}/lib/openjdk|' \
-      -i platform/build-scripts/downloader/src/org/jetbrains/intellij/build/dependencies/JdkDownloader.kt
-    sed \
-      -e 's|BROKEN_PLUGINS_HERE|${./brokenPlugins.json}|' \
-      -i platform/build-scripts/src/org/jetbrains/intellij/build/impl/brokenPlugins.kt
-    sed \
-      -e 's|MAVEN_REPO_HERE|${mvnRepo}/.m2/repository/|' \
-      -e 's|MAVEN_PATH_HERE|${maven}/maven|' \
-      -i build/deps/src/org/jetbrains/intellij/build/impl/BundledMavenDownloader.kt
+    substituteInPlace \
+      platform/build-scripts/src/org/jetbrains/intellij/build/kotlin/KotlinCompilerDependencyDownloader.kt \
+      --replace-fail 'JPS_PLUGIN_CLASSPATH_HERE' '${kotlin-jps-plugin-classpath}'
+    substituteInPlace \
+      platform/build-scripts/src/org/jetbrains/intellij/build/kotlin/KotlinCompilerDependencyDownloader.kt \
+      --replace-fail 'KOTLIN_PATH_HERE' '${kotlin}'
```

Note: if you have multiple substitutions in a single file, these can be chained on a single `substitute` call:

```suggestion
    substituteInPlace \
      platform/build-scripts/src/org/jetbrains/intellij/build/kotlin/KotlinCompilerDependencyDownloader.kt \
      --replace-fail 'JPS_PLUGIN_CLASSPATH_HERE' '${kotlin-jps-plugin-classpath}' \
      --replace-fail 'KOTLIN_PATH_HERE' '${kotlin}'
```